container-reverse-proxy.tf

```diff
@@ -0,0 +1,368 @@
+      "${data.terraform_remote_state.management.outputs.config_bucket.arn}/component/${local.ecs_nginx_rp_config_s3_main_prefix}/*",
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "kms:Decrypt",
+    ]
+
+    resources = [
+      data.terraform_remote_state.management.outputs.config_bucket.cmk_arn,
+    ]
+  }
+}
+
+resource "aws_iam_role" "container_reverse_proxy" {
+  count              = local.reverse_proxy_enabled[local.environment] ? 1 : 0
+  name               = "ReverseProxy"
+  assume_role_policy = data.terraform_remote_state.management.outputs.ecs_assume_role_policy_json
+  tags               = local.common_tags
+}
+
+resource "aws_iam_role_policy" "container_reverse_proxy" {
+  count  = local.reverse_proxy_enabled[local.environment] ? 1 : 0
+  policy = data.aws_iam_policy_document.container_reverse_proxy_read_config[0].json
+  role   = aws_iam_role.container_reverse_proxy[0].id
+}
+
+resource "aws_cloudwatch_log_group" "reverse_proxy_ecs" {
+  count             = local.reverse_proxy_enabled[local.environment] ? 1 : 0
+  name              = "/aws/ecs/main/reverse-proxy"
+  retention_in_days = 30
+  tags              = local.common_tags
+}
+
+resource "aws_ecs_task_definition" "container_reverse_proxy" {
+  count                    = local.reverse_proxy_enabled[local.environment] ? 1 : 0
+  family                   = "nginx-s3"
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  cpu                      = "512"
+  memory                   = "4096"
+  task_role_arn            = aws_iam_role.container_reverse_proxy[0].arn
+  execution_role_arn       = data.terraform_remote_state.management.outputs.ecs_task_execution_role.arn
+
+  container_definitions = <<DEFINITION
+[
+  {
+    "image": "${local.account[local.environment]}.${module.vpc.ecr_dkr_domain_name}/nginx-s3:latest",
+    "name": "squid-s3",
+    "networkMode": "awsvpc",
+    "portMappings": [
+      {
+        "containerPort": ${var.reverse_proxy_http_port}
+      }
+    ],
+    "logConfiguration": {
+      "logDriver": "awslogs",
+      "options": {
+        "awslogs-group": "${aws_cloudwatch_log_group.reverse_proxy_ecs[0].name}",
+        "awslogs-region": "${data.aws_region.current.name}",
+        "awslogs-stream-prefix": "${local.ecs_nginx_rp_config_s3_main_prefix}"
+      }
+    },
+    "placementStrategy": [
+      {
+        "field": "attribute:ecs.availability-zone",
+        "type": "spread"
+      }
+    ],
+    "environment": [
+      {
+        "name": "NGINX_CONFIG_S3_BUCKET",
+        "value": "${data.terraform_remote_state.management.outputs.config_bucket.id}"
+      },
+      {
+        "name": "NGINX_CONFIG_S3_KEY",
+        "value": "${aws_s3_bucket_object.nginx_config[0].key}"
+      }
+    ]
+  }
+]
+DEFINITION
+
+}
+
+resource "aws_ecs_service" "container_internet_proxy" {
+  count           = local.reverse_proxy_enabled[local.environment] ? 1 : 0
+  name            = "container-internet-proxy"
```

Comment:
Would it be better/clearer if this was "reverse-proxy"? Maybe worth defining a local, e.g. 
`reverse_proxy_name = "reverse-proxy"` and then referring to it everywhere the name is needed, like here and in cloudwatch log group name?

Reply:
yea it's meant to be :P